scoutcloud/scoutcloud/src/logic/config/validated.rs

```diff
@@ -0,0 +1,74 @@
+use crate::logic::{
+    config::{types::ConfigValidationContext, variables, Error},
+    ParsedVariableKey, UserVariable,
+};
+use scoutcloud_proto::blockscout::scoutcloud::v1::{
+    DeployConfigInternal, DeployConfigPartialInternal,
+};
+use std::collections::BTreeMap;
+
+#[derive(Default, Debug, Clone, Eq, PartialEq)]
+pub struct ValidatedInstanceConfig {
+    pub vars: BTreeMap<ParsedVariableKey, serde_json::Value>,
+}
+
+macro_rules! parse_config_vars {
+    ($config:ident, $context:ident, $validated_config:ident, $is_partial_config:ident, { $($var:ident),* $(,)? }) => {
+        paste::item! {
+            $({
+                let value: Option<_> = $config.[<$var:snake>].into();
+                let maybe_value = match ($is_partial_config, value) {
+                    (_, Some(value)) => Some(value),
+                    (false, None) => <variables::[<$var:snake>]::[<$var:camel>] as UserVariable<_>>::maybe_default(&$context),
```

Comment:
What will happen if config should be full, but some values which do not have default (e.g., chain_id) will not have a value explicitly set?